### ISSUE TITLE
Disable mode drop down if contnents are invalid json

### ISSUE THREAD

**Issue by yogeshmahajan-1903** [josdejong/jsoneditor#1364]
Steps to reproduce - 
1.Enter valid json content.
2.Delete some semicolon in order to create invalid json.
3.Try to change mode.
4.Console shows error.
<img width="930" alt="Screenshot 2021-08-19 at 2 37 25 PM" src="https://user-images.githubusercontent.com/16016019/130042073-60ecff4d-73ee-4db7-9aaf-c382be6ae11c.png">

Expected - Mode drop down should be disabled if contents are invalid json

**Comment by josdejong** [comment]
In text mode, when the text contents is no valid, parsable JSON, various features cannot be executed: expand, collapse, repair, switch to tree mode, ...

When you try to execute those actions, an error will be thrown. By default, this error is logged in the console only. You can catch the error yourself and do something alternative, like presenting the error to the user in a modal or something like that: 

```js
const options = {
  onError: function (err) {
      alert(err.toString())
    }
}
```

Thinking about it, it may be better default behavior to show such a basic alert to the user on error instead of logging in the console.

In general I try to be careful with disabling buttons: it's easily not clear for a user _why_ a button is disabled and what the user needs to do to get it enabled, so showing an explanatory error to the user when an action cannot be performed has my preference.

**Comment by josdejong** [comment]
I've changed the default behavior to open a basic alert when an error occurs, see cbb95ae

**Comment by yogeshmahajan-1903** [comment]
Showing error will be more useful. 
I have used onError(), but this works only for format & compact json buttons in Code mode.
For other buttons like transform, sort (in cod emode) still error is logged to console.

**Comment by josdejong** [comment]
> For other buttons like transform, sort (in cod emode) still error is logged to console.

Hm, that should not be the case. I'll double check it

**Comment by josdejong** [comment]
You're right. I'll fix that.

**Comment by yogeshmahajan-1903** [comment]
Thanks :)

**Comment by josdejong** [comment]
Fixed via b3b31f2 (not yet published)

**Comment by josdejong** [comment]
Fixes/improvements are published now in `v9.5.4`

**Comment by yogeshmahajan-1903** [comment]
Now error is shown in alert box instead of console.
